tour_model_eval/load_predict.py

```diff
@@ -0,0 +1,92 @@
+import emission.core.get_database as edb
+import emission.analysis.modelling.tour_model.similarity as similarity
+import emission.analysis.modelling.tour_model.data_preprocessing as preprocess
+import jsonpickle as jpickle
+
+
+def loadModelStage(filename):
+    import emission.analysis.classification.inference.mode.seed.pipeline as seedp
+    model = loadModel(filename)
+    return model
+
+
+def loadModel(filename):
+    fd = open(filename, "r")
+    all_model = fd.read()
+    all_model = jpickle.loads(all_model)
+    fd.close()
+    return all_model
+
+
+def in_bin(bin_location_features,new_trip_location_feat,radius):
+    start_b_lon = new_trip_location_feat[0]
+    start_b_lat = new_trip_location_feat[1]
+    end_b_lon = new_trip_location_feat[2]
+    end_b_lat = new_trip_location_feat[3]
+    for feat in bin_location_features:
+        start_a_lon = feat[0]
+        start_a_lat = feat[1]
+        end_a_lon = feat[2]
+        end_a_lat = feat[3]
+        start = similarity.within_radius(start_a_lat, start_a_lon, start_b_lat, start_b_lon,radius)
+        end = similarity.within_radius(end_a_lat, end_a_lon, end_b_lat, end_b_lon, radius)
+        if start and end:
+            continue
+        else:
+            return False
+    return True
+
+
+def predict_labels(trip):
+    radius = 100
+    user = trip[0]['user_id']
+    trip_feat = preprocess.extract_features(trip)[0]
+    trip_loc_feat = trip_feat[0:4]
+    try:
+        # load locations of bins(1st round of clustering)
+        bin_locations = loadModelStage('locations_' + str(user))[0]
+        # load models from the 2nd round of clustering
+        models = loadModelStage('models_' + str(user))[0]
+        # load user labels in all clusters
+        user_labels = loadModelStage('user_labels_' + str(user))[0]
```

Comment:
can you add some information to the PR or the commit about what these data structures look like?

Reply:
added